src/main/java/org/csanchez/jenkins/plugins/kubernetes/AllContainersRunningPodWatcher.java

```diff
@@ -173,4 +219,61 @@ private Pod periodicAwait(int i, long started, long interval, long amount) {
     public PodStatus getPodStatus() {
         return podStatus;
     }
+
+    private boolean isCorrespondingLabels(String taskLabel, String podId, Logger logger) {
+        int taskLabelLen = taskLabel.length();
+        taskLabel = taskLabel.substring(0, taskLabelLen - 2);
+        podId = podId.substring(0, podId.lastIndexOf("-"));
+        //logger.log(INFO,"Comparing: " + taskLabel + " | " + podId);
+        return taskLabel.equals(podId);
+    }
+
+    private String extractLogRecord(List<LogRecord> logs) {
+        StringBuffer msg = new StringBuffer("");
+        for (LogRecord l : logs) {
+            msg.append(l.getLoggerName() + ", ");
+        }
+        return msg.toString();
+    }
+
+    private void writeStream(Run<?, ?> run, String msg) throws IOException {
+        String writeMsg = msg + " \n";
+        FileOutputStream writer = null;
+        try {
+            writer = new FileOutputStream(run.getLogFile().getAbsolutePath(), true);
```

Comment:
writing directly to the file is quite a hack

Reply:
what are the other options do we have to pipe this info to job's console output?

Not all users in our jenkins installation(s) have admin privilege, so not sure if they have access to view the kubernetes logs in UI to determine there is indeed an error in their pod template. Even if we figure out a mechanism to grant them log viewing privilege, its not very intuitive. If we make this info available in console output its right on their face.

Reply:
I know it's a challenge but it's not an easy one to solve last time I've checked

Reply:
yes its a challenge, from what we understand, Jenkins does not provide a handle to cloud plugin to write to the job console output. Without access to such a handle, writing back any node provisioning errors to console is not possible.

Reply:
@jglick is there a way to append to a run log through the API ?

Reply:
@carlossg @jglick Any update on this?

Reply:
I honestly don't know a way to append it to the build log and appending it to the file is something I'm not comfortable with merging. Need somebody with more experience in core to chime in to see if there's an alternative

Reply:
I totally understand appending to the file directly is not an elegant solution. 

@jglick Can you please suggest any alternate mechanism that can be used to pipe the pod creation failure errors to job console output / build log? 

At present, Pod creation error's are not getting displayed in job console output & is a big pain for users. Kubernetes plugin allows user to define the pod template in Jenkins file, however any errors in pod creation is not made available in build log. Most of the time user's don't know wht the build did not start or why it was stuck. Making this information available in Job console output will make it very obvious on why the build did not proceed.

@carlossg @jglick Appreciate your help. We are open for other options / suggestions, please let us know. Addressing this issue will smoothen hundreds of users debug their pod creation issues.

Reply:
Use `stepExecution.getContext().get(TaskListener.class)` to get a stream to which you may print messages associated with the step.

Reply:
@jglick were you able to use the StepExecution method to print pod creation/error messages?

Reply:
Please discuss in the open PR.